lib/pbench/agent/results.py

```diff
@@ -18,39 +18,45 @@ def __init__(self, result_dir, target_dir, user, prefix, config, logger):
         assert (
             config and logger
         ), f"config, '{config!r}', and/or logger, '{logger!r}', not provided"
+        self.result_dir = result_dir
+        self.target_dir = target_dir
+        self.user = user
+        self.prefix = prefix
+        self.config = config
+        self.logger = logger
+
+    def check_result_target_dir(self, result_dir, target_dir):
+        sts = 0
         if not result_dir:
-            logger.error("Result directory not provided")
-            sys.exit(1)
```

Comment:
I don't remember the context of this, but I definitely agree that a utility method like this shouldn't be terminating the process. Strictly within the context of this method, though, after setting `sts` to 1 here and logging the error, we fall through to fail again (redundantly setting `sts` to 1) because `Path(result_dir)` won't resolve. Maybe it makes sense to fully diagnose `target_dir` even if `result_dir` is `None`, but we probably shouldn't go ahead and check the null `result_dir` path when we know it'll fail ...

Reply:
yeah right